pkg/storage/etcd3/store.go

```diff
@@ -277,3 +348,7 @@ func decodeList(elems []*elemForDecode, filter storage.FilterFunc, ListPtr inter
 func notFound(key string) clientv3.Cmp {
 	return clientv3.Compare(clientv3.ModifiedRevision(key), "=", 0)
 }
+
+func sameRevision(key string, rev int64) clientv3.Cmp {
```

Comment:
i feel this is a bad abstraction. Line 353 is more clear since we know we compare the ModRevision.